python_modules/dagster/dagster/_core/definitions/declarative_scheduling/scheduling_context.py

```diff
@@ -73,7 +71,9 @@ class SchedulingContext:
     # which rely on the legact context object. however, this object contains many fields
     # which are not relevant to the scheduling condition evaluation, and so keeping it
     # as a reference here makes it easy to remove it in the future.
-    _legacy_context: AssetConditionEvaluationContext
+    #
+    # marked as Any to avoid circular imports in the pydantic validation logic
+    inner_legacy_context: Any  # AssetConditionEvaluationContext
```

Comment:
Hmmm we discussed the packable value thing but not this. Will this be a permanent state of affairs?

Reply:
@schrockn not permanent, in the sense that all legacy-related stuff will be deleted at some point in the future.

this is also something that seems possible to resolve with a bit of import order tinkering, which I can try to resolve at the top of the stack